### PR TITLE
Improving CI

### LINKED ISSUE #3054: [RFC] Long CI runs
DeepChem CI runs take a long time and it slows down developement time. One of the factors for long CI runs is running unit tests on multiple python versions - for example, unit test on torch CI runs on python 3.7, 3.8, 3.9.

Since unit tests runs same test on different environments, as long as it works in one version, it should also work in another version. I suggest we can run unit test on only one main version (say py 3.8). But we should run build tests on all versions.

### PR DESCRIPTION
**Pull request by arunppsg**
Fix #3054 

Summary of changes:
- formatting ci
    - Earlier: Build only on ubuntu OS python 3.7, tests on ubuntu os python version 3.7, 3.8, 3.9
    - Now: Build on ubuntu OS python 3.8, 3.9, windows py 3.8, tests on ubuntu python 3.7, 3.9
- jax ci
    - Earlier: Build on ubuntu-py 3.7, 3.8, 3.9, run test on ubuntu py 3.7, 3.8, 3.9
    - Now: Build on ubuntu-py 3.7, 3.8, 3.8, run test on ubuntu py 3.7, 3.9
- Tensorflow, torch and common unit tests ci
   - Earlier: Build on ubuntu os in python version 3.7, 3.8, 3.9, run test on ubuntu os in python version 3.7, 3.8, 3.9, windows 3.7
   - Now: Build on ubuntu os in python version 3.7, 3.8, 3.9, windows 3.7 run test on ubuntu os in python version 3.7, 3.9

### CODE REVIEW

**Comment by peastman** [comment]
What python versions do we want to support?  You run some tests on 3.7, but others only on 3.8 and 3.9.  And should we run tests on 3.10 as well?  It's been out for quite a while.

Instead of testing every python version, it's generally ok to just test the earliest and latest supported versions, say 3.7 (or 3.8?) and 3.10.

**Comment by arunppsg** [comment]
> What python versions do we want to support? You run some tests on 3.7, but others only on 3.8 and 3.9. And should we run tests on 3.10 as well? It's been out for quite a while.
We test build on 3.7, 3.8, 3.9 and unit tests on 3.8, 3.9. For 3.10, I am working on it in #2846 - there are some failing dependencies.

> Instead of testing every python version, it's generally ok to just test the earliest and latest supported versions, say 3.7 (or 3.8?) and 3.10.
Makes sense. I will make it as 3.7 and 3.9 here. This makes build test to run on python 3.7, 3.8, 3.9, unit tests on 3.7, 3.9.

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -8,11 +8,15 @@ on:
...
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7]
+        os: [windows-latest, ubuntu-latest]
+        python-version: ['3.8']
+        include:
+        - os: ubuntu-latest
+          python-version: '3.9'
```
Comment (rbharath) [inline .github/workflows/formatting.yml:19]: Should add 3.7

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -39,10 +43,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: [3.7, 3.8, 3.9]
-        include:
-          - os: windows-latest
```
Comment (rbharath) [inline .github/workflows/test.yml:44]: We should add windows back for the unit/model tests, but it's probably ok to remove formatting checks on windows

File: .github/workflows/torch_setup.yml [hunk 3]
```diff
@@ -39,10 +43,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: [3.7, 3.8, 3.9]
-        include:
-          - os: windows-latest
```
Comment (rbharath) [inline .github/workflows/torch_setup.yml:44]: We should keep windows torch/tf/etc checks for correctness

**Review by rbharath — COMMENTED** [review]
Looks good! Once CI is clean, good to merge from my end. @peastman, do you have any more comments on your end?

**Comment by peastman** [comment]
Nothing else, I think it's fine to merge.

**Pull request merged**